### PR TITLE
sail: create new room via http endpoint that returns room id and secret (for later surfacing room id in UI) [ch2294]

### PR DESCRIPTION
**Pull request by maiamcc**
Hello @landism, @dmiller,

Please review the following commits I made in branch maiamcc/http-new-room:

027e8571f2bec3a8f6ff9829f417b4b3c078d72f (2019-04-23 15:44:29 -0400)
server test

7415926efa33303f91682431d54f66b7c11c35cc (2019-04-23 15:29:55 -0400)
count number of NewRoom calls

9a47562ac7be9253c563ea60d2e58aef8df19667 (2019-04-23 15:24:41 -0400)
fake roomer & green tests

1a2066f9f2ca9a000fe79737dc7234cd3d2d10f1 (2019-04-23 15:07:40 -0400)
better test err msg

b8abf912368eee32eb8d532449b5a3dd880edf64 (2019-04-23 15:05:51 -0400)
update make wire to include sail

def7082d0914e682204703e6d9a921d97170ce2f (2019-04-23 12:31:09 -0400)
clean up

333f08faff919e36a887a520fbd538547c1ece3f (2019-04-22 18:02:47 -0400)
send room id and secret with connect request

8bcc73589554c774cd2cca77bb690397651db3e3 (2019-04-22 17:26:14 -0400)
get room id and secret from response

afa2cda00c2e8072c13f73351ccf240360d89b72 (2019-04-22 17:26:09 -0400)
client calls new_room

5ae1dbe05568fb8c9249e93c4b7ea0e700b8fed8 (2019-04-22 17:26:04 -0400)
new_room endpoint

Code review reminders, by giving a LGTM you attest that:

* Commits are adequately tested
* Code is easy to understand and conforms to style guides
* Incomplete code is marked with TODOs
* Code is suitably instrumented with logging and metrics

### CODE REVIEW

File: internal/sail/client/client.go [hunk 1]
```diff
@@ -10,20 +10,23 @@ import (
 	"github.com/windmilleng/tilt/internal/hud/webview"
 	"github.com/windmilleng/tilt/internal/logger"
 	"github.com/windmilleng/tilt/internal/model"
+	sailCommon "github.com/windmilleng/tilt/internal/sail/common"
```
Comment (maiamcc) [inline internal/sail/client/client.go]: this seemed clearer than just referring to it as "common" in the code (and more go-like than naming the package `sail_common`), but idk if this stylistic 🤷‍♀️

File: internal/sail/client/roomer.go [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+	var roomInfo sailCommon.RoomInfo
+	err = json.Unmarshal(body, &roomInfo)
+	if err != nil {
+		return "", "", errors.Wrapf(err, "unmarshaling json: %s", string(body))
+	}
+
+	return roomInfo.RoomID, roomInfo.Secret, nil
+}
+
+func ProvideSailRoomer(addr model.SailURL) SailRoomer {
```
Comment (maiamcc) [inline internal/sail/client/roomer.go]: technically we're denormalizing this address by having it live in both the roomer and the client itself -- is anyone worried about this?

File: internal/sail/server/server.go [hunk 3]
```diff
@@ -73,14 +98,23 @@ func (s SailServer) closeRoom(room *Room) {
...
+		log.Printf("connectRoom: %v", err)
+		return
+	}
+
+	roomID := req.URL.Query().Get(sailCommon.RoomIDKey)
+	secret := req.Header.Get(sailCommon.SecretKey)
+
+	room, err := s.getRoomWithAuth(RoomID(roomID), secret)
+	if err != nil {
+		log.Printf("connectRoom: %v", err)
```
Comment (jazzdan) [inline internal/sail/server/server.go]: Do we want to set HTTP 500 here?

File: internal/sail/common/common.go [hunk 4]
```diff
@@ -0,0 +1,11 @@
+package common
```
[the remark this answers is not in the document]
Comment (maiamcc) [reply internal/sail/common/common.go]: sure that's fine

File: internal/sail/server/server.go (hunk 3 above)
Comment (maiamcc) [reply internal/sail/server/server.go]: ooh good call!

Comment (maiamcc) [reply internal/sail/server/server.go]: ...I don't care enough about parsing the errors to know if it should be 403 or 404 tho, so imma leave a todo

File: internal/sail/server/server.go [hunk 5]
```diff
@@ -34,7 +35,8 @@ func ProvideSailServer(assetServer hudServer.AssetServer) SailServer {
 		assetServer: assetServer,
 	}
 
-	r.HandleFunc("/share", s.startRoom)
+	r.HandleFunc("/new_room", s.newRoom)
```
[the remark this answers is not in the document]
Comment (maiamcc) [reply internal/sail/server/server.go]: 🤷‍♀️ sure!

File: internal/sail/common/common.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (maiamcc) [reply internal/sail/common/common.go]: ...damnit i should just put this in `model/sail.go` shouldn't i? -.-

**Pull request merged**